packages/fx-core/src/component/resource/botService/appStudio/appStudioClient.ts

```diff
@@ -61,25 +65,27 @@ export class AppStudioClient {
   ): Promise<IBotRegistration | undefined> {
     const axiosInstance = AppStudioClient.newAxiosInstance(token);
 
-    const getBotRegistrationResponse: AxiosResponse<any> | undefined = await RetryHandler.Retry(
-      async () => {
-        try {
-          return await axiosInstance.get(`${AppStudioClient.baseUrl}/api/botframework/${botId}`);
-        } catch (e) {
-          if (e.response?.status === 404) {
-            return e.response;
-          } else {
-            e.teamsfxUrlName = "<get-bot-registration>";
-            throw e;
-          }
-        }
-      },
-      true
-    );
-    if (getBotRegistrationResponse?.status === 200) {
-      return <IBotRegistration>getBotRegistrationResponse.data;
-    } else {
-      return undefined;
+    try {
+      const response = await RetryHandler.Retry(() =>
+        axiosInstance.get(`${AppStudioClient.baseUrl}/api/botframework/${botId}`)
+      );
+      if (isHappyResponse(response)) {
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        return <IBotRegistration>response!.data; // response cannot be undefined as it's checked in isHappyResponse.
+      } else {
+        // Defensive code and it should never reach here.
+        throw new Error("Failed to get data");
+      }
+    } catch (e) {
+      if (e.response?.status === HttpStatusCode.NOTFOUND) {
+        return undefined; // Stands for NotFound.
+      } else if (e.response?.status === HttpStatusCode.UNAUTHORIZED) {
+        throw new BotFrameworkNotAllowedToAcquireTokenError();
+      } else {
+        // Potential live site issue cases.
+        e.teamsfxUrlName = "<get-bot-registration>";
```

Comment:
Would be better if we could use constants for all urlNames.

Reply:
Good suggestions. Will do.